storage/watchable_store.go

```diff
@@ -308,6 +310,13 @@ func (s *watchableStore) syncWatchers() {
 		}
 		keyToUnsynced[k][w] = struct{}{}
 	}
+	prefixes := make([]string, len(keyToUnsynced))
+	i := 0
+	for k := range keyToUnsynced {
+		prefixes[i] = k
```

Comment:
we probably should only add key into prefix if w.Prefix is true. Can we move this into the for loop at line 239-310 and only add key into prefix is w.prefix is true? Also can we change prefixes to a map?